simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp

```diff
@@ -227,19 +166,22 @@ class EntityManager
   void broadcastTransform(
     const geometry_msgs::msg::PoseStamped & pose, const bool static_transform = true);
 
-  bool checkCollision(
-    const std::string & first_entity_name, const std::string & second_entity_name);
-
   bool despawnEntity(const std::string & name);
 
-  bool entityExists(const std::string & name);
+  auto isEntitySpawned(const std::string & name) const -> bool;
 
   auto getEntityNames() const -> const std::vector<std::string>;
 
+  auto getEntityOrNullptr(const std::string & name) const
+    -> std::shared_ptr<traffic_simulator::entity::EntityBase>;
```

Comment:
I understand your point of view.
However, I would like to add that there is still the `getEntity` method:
- `getEntity` throws an exception now - where the Entity does not exist - is commonly used in `traffic_simulator`. 
- `getEntityOrNullptr` does not throw exceptions but returns `nullptr` - is mainly used by `simulator_core`, due to the need to handle cases where the Entity does not exist. 

If I change the name of `getEntityOrNullptr` to `getEntityPointer`, then there will be two methods: `getEntityPointer` and `getEntity`, both returning a pointer - which I think is more confusing than `getEntityOrNullptr`+ `getEntity`.

#### Please see the source

https://github.com/tier4/scenario_simulator_v2/blob/4e0c479dd1fd699c0edb891ae9ad47ecc76e037e/simulation/traffic_simulator/src/entity/entity_manager.cpp#L143-L168

I would also like to add that in [PR 5/5](https://github.com/tier4/scenario_simulator_v2/pull/1337) there are changes developed, in which the  `getEntityOrNullptr` method will be completely removed. In parts of the code where it is now necessary, `isEntitySpawned` (`existEntity`) will be used - below I've attached an example. 
This change is included in PR 5/5 because in this PR I focused more on the simulator_core<->api connection itself.

#### Future example - here `getEntityOrNullptr` no longer exist

https://github.com/tier4/scenario_simulator_v2/blob/e4b96cbb89e0a39aaa21fa8b3d576bd4b9de273c/openscenario/openscenario_interpreter/include/openscenario_interpreter/simulator_core.hpp#L584-L675

Reply:
> If I will change the name of getEntityOrNullptr to getEntityPointer, then there will be two methods: getEntityPointer and getEntity, both returning a pointer

My explanation was inadequate. My argument implicitly included the point that the one sending the exception need not return the entity by pointer in the first place. In other words, the function lineup is
- `getEntity(...) const -> EntityBase const&`,
- `getEntity(...) -> EntityBase &` and
- `getEntityPointer(...) -> std::shared_ptr<EntityBase>`.

(Alternatively, the one returning a reference may be more beautiful if it is prefixed with `reference` instead of `get`.)
(The copy constructor and copy assignment operator of EntityBase may need to be removed for safety.)

In this case, would your impression of my opinion differ?

_Note that even if they are completely removed later, I believe that they should be given some reasonable names and roles if they are needed for a transitional period. However, if it requires a great deal of effort to do so, that is a different story._